codes/python/chapter_stack_and_queue/linkedlist_stack.py

```diff
@@ -1,10 +1,84 @@
 '''
 File: linkedlist_stack.py
-Created Time: 2022-11-25
-Author: Krahets (krahets@163.com)
+Created Time: 2022-11-29
+Author: Peng Chen (pengchzn@gmail.com)
 '''
 
-import sys, os.path as osp
+import os.path as osp
+import sys
+
 sys.path.append(osp.dirname(osp.dirname(osp.abspath(__file__))))
 from include import *
 
+""" 基于链表实现的栈 """
+class LinkedListStack:
+    def __init__(self):
+        self.head = None
+
+    """ 获取栈的长度 """
+    def size(self):
+        cnt = 0
+        temp = self.head
+        while temp is not None:
+            temp = temp.next
+            cnt += 1
+        return cnt
+
+    """ 判断栈是否为空 """
+    def is_empty(self):
+        if not self.head.val and not self.head.next:
+            return True
+        return False
+
+    """ 入栈 """
+    def push(self, val):
+        temp = ListNode(val)
+        temp.next = self.head
+        self.head = temp
+
+    """ 出栈 """
+    def pop(self):
+        pop = self.head.val
+        self.head = self.head.next
+        return pop
+
+    """ 访问栈顶元素 """
+    def peek(self):
+        return self.head.val
+
+    def to_array(self):
+        stack = []
+        temp = self.head
+        while temp:
+            stack.insert(0, temp.val)
```

Comment:
It is recommended `append(temp.val)` here, and followed by  `temp.reverse()` to reduce time complexity.

Reply:
The ListNode type seems doed not have the function of reverse(), so I slice the index of it instead, that should be faster than using reverse

Reply:
Got it, I slice the index of it instead, for it should be faster than using reverse(),thanks for comments